### PR TITLE
Pks/update subproject navigation

### PR DESCRIPTION
**Pull request by ninjapanzer**
# Description

Simplifies sub-project index at / README and pushes what details were available into their respective sub-project README.

Adds some backlinks to other projects that act as dependencies.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] This change is only to documentation

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Make sure links work

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,94 +16,14 @@ Please note that the data for GSD is `CC0`, everything else is `Apache v2.0`. Se
...
+
+- [GSD Analysis](/gsd-analysis#gsd-analysis "A review of the GSD data using a python parser")
+- [GSD API](/gsd-api#gsd-api "REST API for interfacing with the GSD Database")
+- [GSD Bot](/gsd-bot#gsd-bot "Automation around specifically formatted issues within the gsd-database repo")
+- [GSD GitHub Actions](/gsd-github-actions#github-actions "GSD Github Actions")
+- [GSD Requests](/gsd-requests#webform "Webform for submitting new GSD IDs")
+- [GSD Schema](/gsd-schema#gsd-schema "JSON schema of a GSD identifier")
+- [GSD URL Processing](/gsd-url-processing#gsd-url-processing)
+- [GSD Web](/gsd-web#gsd-web "A web based user interface for the GSD data")
+- [Local Scripts](/local-scripts#local-scripts)
```
Comment (joshbuker) [inline README.md:29]: This format provides the one-line description when you hover, but I suspect that might not be intuitive for someone viewing this for the first time. I do like the simplified format though.

**Review by joshbuker — APPROVED** [review]
There's still room for improvement, but this is a great step in the right direction. Thanks!

File: README.md (hunk 1 above)
Comment (ninjapanzer) [reply README.md:29]: This isn't expressly for browser users. It's for screen readers and good indexing. If you think there should be a contextual description for these links to help describe them I don't mind putting them back.

Comment (joshbuker) [reply README.md:29]: Ah, that makes sense.

It's fine for now, we can also add the descriptions back if folks mention it's confusing without them.

**Pull request merged**